server/src/internalClusterTest/java/org/opensearch/index/ShardIndexingPressureSettingsIT.java

```diff
@@ -0,0 +1,827 @@
+            .put(ShardIndexingPressureMemoryManager.THROUGHPUT_DEGRADATION_LIMITS.getKey(), 1)
+            .build();
+        restartCluster(settings);
+
+        assertAcked(prepareCreate(INDEX_NAME, Settings.builder()
+            .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, 1)
+            .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 1)));
+        ensureGreen(INDEX_NAME);
+
+        Tuple<String, String> primaryReplicaNodeNames = getPrimaryReplicaNodeNames(INDEX_NAME);
+        String primaryName = primaryReplicaNodeNames.v1();
+        String replicaName = primaryReplicaNodeNames.v2();
+        String coordinatingOnlyNode = getCoordinatingOnlyNode();
+
+        IndexService indexService = internalCluster().getInstance(IndicesService.class, primaryName).iterator().next();
+        Index index = indexService.getIndexSettings().getIndex();
+        ShardId shardId= new ShardId(index, 0);
+
+        boolean randomBoolean = randomBoolean();
+
+        // Send first request which gets successful
+        ActionFuture<BulkResponse> successFuture;
+        if (randomBoolean) {
+            successFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+        } else {
+            successFuture = client(primaryName).bulk(bulkRequest);
+        }
+        successFuture.actionGet();
+
+        final CountDownLatch replicationSendPointReached = new CountDownLatch(1);
+        final CountDownLatch latchBlockingReplicationSend = new CountDownLatch(1);
+
+        TransportService primaryService = internalCluster().getInstance(TransportService.class, primaryName);
+        final MockTransportService primaryTransportService = (MockTransportService) primaryService;
+        TransportService replicaService = internalCluster().getInstance(TransportService.class, replicaName);
+        final MockTransportService replicaTransportService = (MockTransportService) replicaService;
+
+        primaryTransportService.addSendBehavior((connection, requestId, action, request, options) -> {
+            if (action.equals(TransportShardBulkAction.ACTION_NAME + "[r]")) {
+                try {
+                    replicationSendPointReached.countDown();
+                    latchBlockingReplicationSend.await();
+                } catch (InterruptedException e) {
+                    throw new IllegalStateException(e);
+                }
+            }
+            connection.sendRequest(requestId, action, request, options);
+        });
+
+        final ThreadPool replicaThreadPool = replicaTransportService.getThreadPool();
+        final Releasable replicaRelease = blockReplicas(replicaThreadPool);
+        // Send one more requests which remains outstanding for delayed time
+        if (randomBoolean) {
+            successFuture = client(coordinatingOnlyNode).bulk(bulkRequest);
+        } else {
+            successFuture = client(primaryName).bulk(bulkRequest);
+        }
+        // Delay to breach the success time stamp threshold
+        Thread.sleep(3000);
```

Comment:
How are we deciding on sleeps

Reply:
Default value for ``SUCCESSFUL_REQUEST_ELAPSED_TIMEOUT`` is ``3000ms``. Here we are introducing the sequential delay to mimic delay in completion of the request.